### PR TITLE
allow building for any specified $OS

### PR DESCRIPTION
**Pull request by zmiklank**
Partly fixes https://github.com/sclorg/s2i-base-container/issues/279.

### CODE REVIEW

**Comment by zmiklank** [comment]
[test]

**Comment by zmiklank** [comment]
@phracek Do we still use the *-candidate $OSes? If yes I will rework this PR to work with them. Current proposal supports only standard targets (rhel7, rhel8, fedora, ...) for which the Dockerfiles exist.

File: build.sh [hunk 1]
```diff
@@ -212,22 +212,10 @@ echo "Built versions are: $dirs"
...
-  elif [ "$OS" == "fedora" ] || [ "$OS" == "fedora-candidate" ]; then
-    docker_build_with_version Dockerfile.fedora
-  elif [ "$OS" == "centos6" ] || [ "$OS" == "centos6-candidate" ]; then
-    docker_build_with_version Dockerfile.centos6
-  elif [ "$OS" == "c8s" ] || [ "$OS" == "c8s-candidate" ]; then
-    docker_build_with_version Dockerfile.c8s
-  elif [ "$OS" == "c9s" ] || [ "$OS" == "c9s-candidate" ]; then
-    docker_build_with_version Dockerfile.c9s
-  else
+  if [ "$OS" == "centos7" ] || [ "$OS" == "centos7-candidate" ]; then
```
Comment (phracek) [inline build.sh:215]: I don't know why `-candidate` were used?. Even `centos7` can be used like the rest.
I guess only one row is needed `docker_build_with_version Dockerfile."$OS"`

I guess we alreade have a Dockerfile as symlink to Dockerfile.centos7, right?

**Review by phracek — CHANGES_REQUESTED** [review]
Thanks for this improvement.

See me comment.

File: build.sh (hunk 1 above)
Comment (zmiklank) [reply build.sh:215]: > I guess we alreade have a Dockerfile as symlink to Dockerfile.centos7, right?

AFAIK this is valid only for Python container. So we need to support centos7 separately.

Comment (zmiklank) [reply build.sh:215]: @pkubatrh Do you know, if *-candidate targets are still used somewhere, please?

**Review by phracek — APPROVED** [review]
LGTM. Let's get merge it.

**Pull request merged**